src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_TupleBinaryOperator.cs

```diff
@@ -0,0 +1,410 @@
+                if (expr.Kind != BoundKind.Conversion)
+                {
+                    return expr;
+                }
+
+                var conversion = (BoundConversion)expr;
+                if (!conversion.Conversion.IsImplicit || !conversion.Conversion.IsNullable)
+                {
+                    return expr;
+                }
+
+                expr = conversion.Operand;
+            }
+        }
+
+        private BoundExpression MakeTemp(BoundExpression loweredExpression, ArrayBuilder<LocalSymbol> temps, ArrayBuilder<BoundExpression> effects)
+        {
+            BoundLocal temp = _factory.StoreToTemp(loweredExpression, out BoundAssignmentOperator assignmentToTemp);
+            effects.Add(assignmentToTemp);
+            temps.Add(temp.LocalSymbol);
+            return temp;
+        }
+
+        /// <summary>
+        /// Returns a temp which is initialized with lowered-expression.HasValue
+        /// </summary>
+        private BoundExpression MakeHasValueTemp(BoundExpression expression, ArrayBuilder<LocalSymbol> temps,
+            ArrayBuilder<BoundExpression> effects)
+        {
+            BoundExpression hasValueCall = MakeNullableHasValue(expression.Syntax, expression);
+            return MakeTemp(hasValueCall, temps, effects);
+        }
+
+        /// <summary>
+        /// Returns a temp which is initialized with lowered-expression.GetValueOrDefault()
+        /// </summary>
+        private BoundExpression MakeValueOrDefaultTemp(BoundExpression expression,
+            ArrayBuilder<LocalSymbol> temps, ArrayBuilder<BoundExpression> effects)
+        {
+            BoundExpression valueOrDefaultCall = MakeOptimizedGetValueOrDefault(expression.Syntax, expression);
+            return MakeTemp(valueOrDefaultCall, temps, effects);
+        }
+
+        /// <summary>
+        /// Produces a chain of equality (or inequality) checks combined logically with AND (or OR)
+        /// </summary>
+        private BoundExpression RewriteNonNullableNestedTupleOperators(TupleBinaryOperatorInfo.Multiple operators,
+            BoundExpression left, BoundExpression right, TypeSymbol type,
+            ArrayBuilder<LocalSymbol> temps, ArrayBuilder<BoundExpression> effects, TupleOperatorSideEffectsAndTemps effectAndTemps, BinaryOperatorKind operatorKind, bool inLeftLiteral, bool inRightLiteral)
+        {
+            Debug.Assert(left.Type?.IsNullableType() == false);
+            Debug.Assert(right.Type?.IsNullableType() == false);
+
+            ImmutableArray<TupleBinaryOperatorInfo> nestedOperators = operators.Operators;
+
+            BoundExpression currentResult = null;
+            for (int i = 0; i < nestedOperators.Length; i++)
+            {
+                BoundExpression leftElement = GetTuplePart(left, i, nestedOperators, temps, effects, effectAndTemps, isRight: false);
+                BoundExpression rightElement = GetTuplePart(right, i, nestedOperators, temps, effects, effectAndTemps, isRight: true);
```

Comment:
Perhaps pass `effectsAndTemps.temps` and `effectsAndTemps.leftInit` directly and remove `isRight`.